stdlib/public/core/Integers.swift.gyb

```diff
@@ -283,8 +340,8 @@ public func != <T : BinaryInteger, U : BinaryInteger>(lhs:T, rhs: U) -> Bool {
 
 @_transparent
 public func < <T : BinaryInteger, U : BinaryInteger>(lhs: T, rhs: U) -> Bool {
-  let lhsSign = lhs < 0 ? -1 : lhs > 0 ? 1 : 0
-  let rhsSign = rhs < 0 ? -1 : rhs > 0 ? 1 : 0
+  let lhsSign = lhs < (0 as T) ? -1 : lhs > (0 as T) ? 1 : 0
+  let rhsSign = rhs < (0 as U) ? -1 : rhs > (0 as U) ? 1 : 0
```

Comment:
If these are ambiguous without the casts, we should make it a requirement of BinaryInteger that there be a (Self,Self) version of the comparison.